File: src/main/java/um/tesoreria/core/model/dto/UMPreferenceMPDto.java
package um.tesoreria.core.model.dto;

import lombok.*;
import um.tesoreria.core.hexagonal.mercadoPagoContext.domain.model.MercadoPagoContext;
import um.tesoreria.core.hexagonal.chequeraCuota.infrastructure.persistence.entity.ChequeraCuotaEntity;
import um.tesoreria.core.util.Jsonifier;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class UMPreferenceMPDto {

    private MercadoPagoContext mercadoPagoContext;
    private ChequeraCuotaEntity chequeraCuota;

    public String jsonify() {
        return Jsonifier.builder(this).build();
    }
}

File: src/main/java/um/tesoreria/core/kotlin/model/dto/UMPreferenceMPDto.kt
package um.tesoreria.core.kotlin.model.dto

import um.tesoreria.core.kotlin.model.ChequeraCuota
import um.tesoreria.core.kotlin.model.MercadoPagoContext

data class UMPreferenceMPDto(
    var mercadoPagoContext: MercadoPagoContext? = null,
    var chequeraCuota: ChequeraCuota? = null
) {
    // Builder class
    class Builder {
        private var mercadoPagoContext: MercadoPagoContext? = null
        private var chequeraCuota: ChequeraCuota? = null

        fun mercadoPagoContext(mercadoPagoContext: MercadoPagoContext?) = apply {
            this.mercadoPagoContext = mercadoPagoContext
        }

        fun chequeraCuota(chequeraCuota: ChequeraCuota?) = apply {
            this.chequeraCuota = chequeraCuota
        }

        fun build() = UMPreferenceMPDto(
            mercadoPagoContext = mercadoPagoContext,
            chequeraCuota = chequeraCuota
        )
    }

    companion object {
        fun builder() = Builder()
    }
}
